processor/src/main/java/org/derive4j/processor/DeriveConfigBuilder.java

```diff
@@ -217,12 +217,13 @@ private Stream<Function<DeriveConfig, DeriveConfig>> annotationConfig(TypeElemen
   }
 
   private Stream<Function<DeriveConfig, DeriveConfig>> deriveConfigs(TypeElement typeElement, Element element,
-      HashSet<AnnotationMirror> seenAnnotations) {
-    return element.getAnnotationMirrors().stream().sequential().filter(a -> !seenAnnotations.contains(a)).flatMap(a -> {
-      seenAnnotations.add(a);
-      return concat(deriveConfigs(typeElement, a.getAnnotationType().asElement(), seenAnnotations),
-          annotationConfig(typeElement, a));
-    });
+      HashSet<Map<? extends ExecutableElement, ? extends AnnotationValue>> seenAnnotations) {
```

Comment:
This is a bit of a weird HashSet, but hashCode is well defined for Map, ExecutableElement and AnnotationValue so it should just work.